### PR TITLE
delete replicaset when delete namespace

### LINKED ISSUE #21049: Can't delete replicaset
The delete command for a replicaset hangs forever.  This then manifests as phantom RSs when I delete and recreate a namespace.

```
thockin@freakshow:~/src/th-kube-demos$ k delete ns demos
namespace "demos" deleted
thockin@freakshow:~/src/th-kube-demos$ kubectl --namespace=demos get replicaset
CONTROLLER              CONTAINER(S)   IMAGE(S)   SELECTOR                                                                      REPLICAS   AGE
deployment-demo-4l559   busybox        busybox    demo in (deployment),demo-version in (v1),pod-template-hash in (4197825882)   0          1h
                        nginx          nginx                                                                                    
deployment-demo-ftv0k   busybox        busybox    demo in (deployment),pod-template-hash in (2688726913)                        0         1h
                        nginx          nginx                                                                                    
deployment-demo-okbti   busybox        busybox    demo in (deployment),pod-template-hash in (2809771906)                        3         1h
                        nginx          nginx                                                                                    
deployment-demo-p514c   busybox        busybox    demo in (update),demo-version in (v1),pod-template-hash in (2763045788)       3         1h
                        nginx          nginx                       
```

### PR DESCRIPTION
**Pull request by mqliang**
partially fixes https://github.com/kubernetes/kubernetes/issues/21049

@madhusudancs @thockin

### CODE REVIEW

**Comment by thockin** [comment]
The problems in #21049 is that even a manual delete by name is failing

@derekwaynecarr

**Comment by derekwaynecarr** [comment]
Need to update the test case.

see: https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/namespace/namespace_controller_test.go#L119

File: pkg/controller/namespace/namespace_controller_utils.go [hunk 1]
```diff
@@ -184,6 +184,12 @@ func deleteAllContent(kubeClient clientset.Interface, versions *unversioned.APIV
 				return estimate, err
 			}
 		}
+		if containsResource(resources, "replicasets") {
```
Comment (mqliang) [inline pkg/controller/namespace/namespace_controller_utils.go]: This if statement will return false so ` deleteReplicaSets` will never run, have no idea what's going wrong.

Comment (derekwaynecarr) [inline pkg/controller/namespace/namespace_controller_utils.go]: Are you referring to the test case or when running the actual code?

The test case is mocked up, so you need to add your resource:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/namespace/namespace_controller_test.go#L151

**Comment by derekwaynecarr** [comment]
Make sure you are testing with this PR:
https://github.com/kubernetes/kubernetes/pull/21035

**Comment by mqliang** [comment]
@derekwaynecarr 
> Make sure you are testing with this PR:

Oh, I think I should rebase.

**Comment by derekwaynecarr** [comment]
@thockin - taking this off your hands.  @mqliang i am on slack/irc if you need help.

**Comment by bgrant0607** [comment]
cc @kargakis @janetkuo

**Comment by derekwaynecarr** [comment]
lgtm

**Comment by derekwaynecarr** [comment]
looks like this flake: https://github.com/kubernetes/kubernetes/issues/21094

**Comment by derekwaynecarr** [comment]
@k8s-bot test this issue: #21094

**Comment by derekwaynecarr** [comment]
@k8s-bot test this issue: #21094

**Pull request merged**